### ISSUE TITLE
include_unordered_json does not take missing items into account

### ISSUE THREAD

**Issue by jeroenj** [waterlink/rspec-json_expectations#22]
If the expected Array has less values than the actual Array the matcher succeeds:

```ruby
expect(%(["foo", "bar", "baz"])).to include_unordered_json(["foo", "bar"]) # this will pass
```

/cc @esjee (since it was your contribution in #17)

**Comment by jeroenj** [comment]
I might try to look into this this week if I find the time.

**Comment by esjee** [comment]
Oh no! Nice find! I'll also try to find time for this if you don't beat me to it. I'll post a message here if I pick it up, to avoid wasting your time.

**Comment by jeroenj** [comment]
Great. I'll do that too. I have quite some work on my plate this week (and the following weeks). We've worked around it for now, but I'm still planning to fix it. :)

**Comment by esjee** [comment]
Upon closer look, I'm wondering if this isn't expected behavior for something that "includes". It looks like you want stricter matching, similar to `match_array`/`contain_exactly`.

What do you think of adding a `match_unordered_json` matcher that does this?

```ruby
# Pass
expect(%(["foo", "bar", "baz"])).to include_unordered_json(["foo", "bar"])  # rspec-json_expectations
expect(["foo", "bar", "baz"]).to include("foo", "bar")  # built-in matcher

# Fail
expect(%(["foo", "bar", "baz"])).to match_unordered_json(["foo", "bar"])  # example, this matcher does not exist
expect(["foo", "bar", "baz"]).to contain_exactly("foo", "bar")  # built-in matcher
expect(["foo", "bar", "baz"]).to match_array(["foo", "bar"]) # built-in matcher
```

@jeroenj What do you think?

**Comment by jeroenj** [comment]
Looks like a good idea but it might add some confusion. Maybe the naming wasn't optimal to start with. I haven't checked but I'd say that `match_json` and `match_unordered_json` would have been more clear to start with.

I'd see what @waterlink has to say about it. Having `match_unordered_json` next to `include_unordered_json` would definately be a step in the right direction. 👍

**Comment by esjee** [comment]
Hey @jeroenj, just wanted to say I'm trying to create a PR that implements `match_unordered_json`. While you may expect some rough edges, I figured a PR would be a good way to get @waterlink's thoughts on this, and see where can go.

**Comment by jeroenj** [comment]
Awesome. I didn't got the time yet to look into it, but I'll give your PR a try when it's available. 👍

**Comment by waterlink** [comment]
@jeroenj @esjee I have released this as a part of `2.0.0` version
